addons/io/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitAccessoryRegistry.java

```diff
@@ -0,0 +1,102 @@
+package org.openhab.io.homekit.internal;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.openhab.io.homekit.internal.accessories.GroupedAccessory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.beowulfe.hap.HomekitAccessory;
+import com.beowulfe.hap.HomekitRoot;
+
+/**
+ * Stores the created HomekitAccessories. GroupedAccessories are also held here
+ * in a pre-created pending state until all required characteristics are found.
+ *
+ * @author Andy Lintner
+ */
+class HomekitAccessoryRegistry {
```

Comment:
Ok, fine for me. It is just a "hardly ever used practice", but you are right that reducing visibility to what is needed is a good thing.